### PR TITLE
Default XmlPeek, XmlPoke, and XslTransformation to DtdProcessing.Prohibit

### LINKED ISSUE #5817: XmlPoke Should Use DtdProcessing.Prohibit
See https://github.com/dotnet/msbuild/issues/5810 for a bit of context.

There's an inconsistency with `XmlPoke` and `XslTransformation`. The [docs on XmlReaderSettings.DtdProcessing](https://docs.microsoft.com/dotnet/api/system.xml.xmlreadersettings.dtdprocessing) state that the default value is `Prohibit`, which neither `XmlPoke` or XslTransformation` follow.

### PR DESCRIPTION
**Pull request by vijaya-lakshmi-venkatraman**
Fixes #5817 

### Context
XMLReaderSettings has default value of _Prohibit_ while XMLPeek, XMLPoke and XSLTransformation files have _Ignore_ set. This has been changed to _Prohibit_

### Changes Made
Changed the default value of XMLReaderSettings from Ignore to Prohibit

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=6418) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=6418) before we can accept your contribution.<br/><br/>:x: vijaya-lakshmi-venkatraman [sign now](https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=6418)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/msbuild?pullRequest=6418) it.</sub>

File: src/Tasks/XmlPeek.cs
Comment (BenVillalobos) [inline src/Tasks/XmlPeek.cs:317]: This parameter is set by the `ProhibitDtd` property (line ~71), we should also default that value to true.

**Review by BenVillalobos — COMMENTED** [review]
🥳 Looks good! See my other comment for what looks to be the final touch-up on this PR.

File: src/Tasks/XmlPeek.cs
Comment (vijaya-lakshmi-venkatraman) [reply src/Tasks/XmlPeek.cs:317]: Thanks,
So I will add the below (assuming we are on C# 6.0).
```
public bool ProhibitDtd { get; set; } = true;
```
Is that right?

Comment (vijaya-lakshmi-venkatraman) [reply src/Tasks/XmlPeek.cs:317]: Thank you for your input. 
I will remove the optional value for _prohibitDtd_ parameter.

**Comment by vijaya-lakshmi-venkatraman** [comment]
> Thank you for the contribution! Do you think it would be useful for `XmlPoke` and `XslTransformation` to have the `ProhibitDtd` property, similar to `XmlPeek`, so developers can switch to the old behavior if needed?

👍  
I can include that property in XMLPoke and XSLTransformation.
Do I also have to make changes similar to this in these files so the value is picked up based on what's set on the property?
`DtdProcessing = prohibitDtd ? DtdProcessing.Prohibit : DtdProcessing.Ignore`

**Comment by ladipro** [comment]
> Do I also have to make changes similar to this in these files so the value is picked up based on what's set on the property?
`DtdProcessing = prohibitDtd ? DtdProcessing.Prohibit : DtdProcessing.Ignore`

I'll defer to @BenVillalobos to confirm. In general, I think that we should strive for
1. Consistency. By default, all these tasks should behave the same with respect to DTD processing.
2. Backward compatibility. Unless we are 100% sure that the previous setting was useless, there should be a way for users to bring it back if they depend on it.

In spirit of 2., is it safe to unconditionally change the behavior when loading namespaces (in `XmlPeek` and `XmlPoke`) and XSLT parameters (in `XsltTransform`)?

**Comment by BenVillalobos** [comment]
> Backward compatibility. Unless we are 100% sure that the previous setting was useless, there should be a way for users to bring it back if they depend on it.

The classic msbuild 'gotcha'. It's public so _theoretically_ tons of devs could be using it, but we have no way to tell and should assume worst case scenario and provide an opt-into original behavior. Giving XmlPoke and XslTransformation a public `ProhibitDtd` boolean (defaulted to true) is a good idea in that respect.

So to directly answer this:
> Do I also have to make changes similar to this in these files so the value is picked up based on what's set on the property?
DtdProcessing = prohibitDtd ? DtdProcessing.Prohibit : DtdProcessing.Ignore

Yep! Each class should have the `ProhibitDtd` boolean and have them set DtdProcessing to either Prohibit or Ignore based on that boolean.

File: src/Tasks/XmlPeek.cs
Comment (Forgind) [reply src/Tasks/XmlPeek.cs:317]: > Thanks,
> So I will add the below (assuming we are on C# 6.0).
> 
> ```
> public bool ProhibitDtd { get; set; } = true;
> ```
> 
> Is that right?

Yes, that looks right. So change line 71 to that and remove this default value, and we're good!

**Review by Forgind — APPROVED** [review]
Thanks for fixing this!

**Comment by Forgind** [comment]
@vijaya-lakshmi-venkatraman, do you know when you'll have a chance to get back to this?

**Comment by vijaya-lakshmi-venkatraman** [comment]
> @vijaya-lakshmi-venkatraman, do you know when you'll have a chance to get back to this?

@Forgind Apologies for not updating you. I have been trying to get approval from my organization for signing the CLA (which seems to be required for merging the changes). Please can I get a week's time?

**Comment by Forgind** [comment]
@vijaya-lakshmi-venkatraman Of course! I'm just making sure everything open is still in progress. Take all the time you need.

**Comment by Forgind** [comment]
Hi @vijaya-lakshmi-venkatraman, is this still on your radar? Has your organization said anything positive or negative on signing the CLA?

**Comment by vijaya-lakshmi-venkatraman** [comment]
> Hi @vijaya-lakshmi-venkatraman, is this still on your radar? Has your organization said anything positive or negative on signing the CLA?

My deepest apologies to keep this open for so long. 
Yes! There has been some pointers on who can sign the CLA. Unfortunately the point of contact to authorize my CLA has been off sick. I will need a little more time to sort this admin part.

Reg the code changes, the changes required is
1) public bool ProhibitDtd { get; set; } = true; on all files
2) Remove default value of true in the methods
Is there anything else needed?

**Comment by Forgind** [comment]
No worries! I hope your contact feels better (and I hope it isn't covid).

I think that's right, though I haven't looked at this in a while, so I don't remember for sure.

**Comment by rainersigwald** [comment]
While you're getting approval I'm going to convert this to draft so it's more obvious that it's not ready yet.

**Comment by Forgind** [comment]
If you deleted your fork, it's probably easier just to make a new PR with this content. Then the bot will pop up and let you sign from there.

**Comment by vijaya-lakshmi-venkatraman** [comment]
Thank you
I am able to see the contents of the CLA
Can I just request the authorized signatory of our organization sign the CLA on the link (eg: https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=6418) or is there another process for Corporate CLA?

**Comment by rainersigwald** [comment]
@terrajobst do you know how to advise @vijaya-lakshmi-venkatraman on this question?

> Can I just request the authorized signatory of our organization sign the CLA on the link (eg: https://cla.dotnetfoundation.org/dotnet/msbuild?pullRequest=6418) or is there another process for Corporate CLA?

**Comment by vijaya-lakshmi-venkatraman** [comment]
@terrajobst Please can you help with the Corporate CLA process?

**Pull request closed without merging**
